lib/manageiq_performance/middlewares/active_record_queries.rb

```diff
@@ -153,12 +153,29 @@ def skip_schema_queries?
         IGNORED_PAYLOAD    = %w[EXPLAIN CACHE].freeze
 
         def ignore_payload?(payload)
-          payload[:exception] ||
+          payload[:exception] || payload[:cached] ||
           (skip_schema_queries? and SCHEMA_QUERY_TYPES.include?(payload[:name])) ||
           IGNORED_PAYLOAD.include?(payload[:name])
         end
       end
 
+      def self.clear_memoized_logger_vars
+        %w(sql.active_record instantiation.active_record).each do |event|
+          listeners = ActiveSupport::Notifications.notifier.listeners_for(event)
+          logger    = listeners.detect do |l|
+            l.instance_variable_get(:@delegate).is_a? self::Logger
+          end.instance_variable_get(:@delegate)
+
+          [
+            :@include_queries, :@include_trace, :@skip_schema_queries
```

Comment:
This is throwing me a little.
Seems right, just not wrapping my mind around what exactly is happening. Or maybe it is the why part that is getting me

Reply:
@kbrock I will give an example:

Say you are in an `irb` session, and you call:

```ruby
irb> ManageIQPerformance.profile { some_method }
```

That will [call `active_record_queries_initialize`](https://github.com/ManageIQ/manageiq-performance/blob/b5706d89/lib/manageiq_performance/middlewares/active_record_queries.rb#L7-L18) as part of the `ManageIQPerformance::Middleware.new` method.  First time, it will subscribe `Middleware:: ActiveRecordQueries::Logger` to the `ActiveSupport::Notifications` for `%w(sql.active_record instantiation.active_record)`.

Then if you call:

```ruby
irb> ManageIQPerformance.profile { other_method }
```

It creates a new `ManageIQPerformance::Middleware.new`, which re-calls `active_record_queries_initialize`.  The second time, though, it doesn't add new subscribers, since they already exist, and the `Middleware` uses `Thread` variables to make sure it is recording or not recording when the block is finished, so new one's aren't required.  We also make sure we aren't making duplicate subscribers here, otherwise we would get double of the results saved in our middleware storage.

But this means that if you wanted to include stack_traces on the first one, but not on the second (using the `.with_config` that is part of `.profile`), it wouldn't let you since the `@include_trace` is memoized on the first call of [`include_trace?`](https://github.com/ManageIQ/manageiq-performance/blob/b5706d898/lib/manageiq_performance/middlewares/active_record_queries.rb#L140-L143), and will never re-check the settings.

Make sense or did I over look something in the explanation?

Reply:
sounds good. thnx
(think I knew some of this at one time, but needed a refresher)